beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -456,3 +461,110 @@ func getBlockFields(b interfaces.ReadOnlySignedBeaconBlock) logrus.Fields {
 		"version":       b.Block().Version(),
 	}
 }
+
+// detectAndBroadcastEquivocation checks if the given block is an equivocating block by comparing it with
+// the head block. If the blocks are from the same slot and proposer but have different signatures,
+// it verifies the difference constitutes a slashable offense before creating and broadcasting a proposer
+// slashing object.
+func (s *Service) detectAndBroadcastEquivocation(ctx context.Context, blk interfaces.ReadOnlySignedBeaconBlock) error {
+	slot := blk.Block().Slot()
+	proposerIndex := blk.Block().ProposerIndex()
+
+	// Get head block for comparison
+	headBlock, err := s.cfg.chain.HeadBlock(ctx)
+	if err != nil {
+		return errors.Wrap(err, "could not get head block")
+	}
+
+	// Only proceed if this block is from same slot and proposer as head
+	if headBlock.Block().Slot() != slot || headBlock.Block().ProposerIndex() != proposerIndex {
+		return nil
+	}
+
+	// Compare signatures
+	sig1 := blk.Signature()
+	sig2 := headBlock.Signature()
+
+	// If signatures match, these are the same block
+	if sig1 == sig2 {
+		return nil
+	}
+
+	header1, err := blk.Header()
+	if err != nil {
+		return errors.Wrap(err, "could not get header from new block")
+	}
+	header2, err := headBlock.Header()
+	if err != nil {
+		return errors.Wrap(err, "could not get header from head block")
+	}
+
+	slashing := &ethpb.ProposerSlashing{
+		Header_1: header1,
+		Header_2: header2,
+	}
+
+	// Verify basic slashing conditions before state check
+	if err := verifySlashableBlock(slashing); err != nil {
+		return errors.Wrap(err, "block headers not slashable")
+	}
+
+	// Get state for final verification
+	headState, err := s.cfg.chain.HeadStateReadOnly(ctx)
+	if err != nil {
+		return errors.Wrap(err, "could not get head state")
+	}
+
+	// Verify the slashing against current state
```

Comment:
I may have been wrong in my previous review and if that is the case I apologize, but the function `VerifyProposerSlashing` already performs all the checks in `verifySlashableBlock` so I suggest the following changes
- Remove the function`verifySlashableBlock` 
- just call `VerifyProposerSlashing` after making sure that the signatures are different. 

However, there is a little care that needs to be taken. If `VerifyProposerSlashing` returns an error because the signature failed to verify, we need to return `PubsubReject` from the main block validation pipeline. 

The PR and it's logic looks good to me and I think it's safe to merge after this change.

Reply:
Thanks for the feedback, Just pushed the changes